kola/tests/ignition/oem.go

```diff
@@ -47,18 +50,46 @@ storage:
           set linux_append="flatcar.autologin"
           # Needed if --qemu-skip-mangle is not set
           set linux_console="console=ttyS0,115200"
+`),
+	})
+	register.Register(&register.Test{
+		Name:        "cl.ignition.oem.regular",
+		Run:         reusePartitionNew,
+		ClusterSize: 1,
+		Distros:     []string{"cl"},
+		// This test overwrites the grub.cfg which does not work on cloud environments after reboot
+		Platforms:  []string{"qemu", "qemu-unpriv"},
+		MinVersion: semver.Version{Major: 3603},
+		UserData: conf.Butane(`---
+variant: flatcar
+version: 1.0.0
+storage:
+  filesystems:
+     - device: "/dev/disk/by-label/OEM"
+       format: "btrfs"
+       path: /oem
+  files:
+    - path: /oem/grub.cfg
+      mode: 0644
+      overwrite: true
+      contents:
+        inline: |
+          set linux_append="flatcar.autologin"
+          # Needed if --qemu-skip-mangle is not set
+          set linux_console="console=ttyS0,115200"
 `),
 	})
 	register.Register(&register.Test{
 		// Check new behavior from https://github.com/flatcar/bootengine/pull/58
 		// to not have to specify the OEM filesystem in Ignition
-		Name:        "cl.ignition.oem.indirect",
-		Run:         reusePartition,
+		Name:        "cl.ignition.oem.indirect.old",
```

Comment:
I mean, we can have two tests but we don't need to limit the version of the old test, it's even better to keep it to check that we are compatible with older configs.

Reply:
Kept the duplicated tests, but dropped the `EndVersion`, so both variants will run on a new enough Flatcar version. Changed the names too, (so instead of `foo.old` and `foo`, there's now `foo` and `foo.new`).

Reply:
Updated the changelog too.